modules/agent-install-networking.adoc

```diff
@@ -0,0 +1,152 @@
+// Module included in the following assemblies:
+//
+// * installing/installing-with-agent/installing-with-agent.adoc
+
+:_content-type: CONCEPT
+[id="agent-install-networking_{context}"]
+= About networking
+
+During the initial boot, the machines require an IP address configuration that is set either through a Dynamic Host Configuration Protocol (DHCP) server or statically by
+selecting the below options. If you provide the IP address for the `rendezvousIP` field through the DHCP option, then ensure that it is for the machine's IP that is going to be used for deployment.
+This IP address is required for a node to be identified as ** node 0 **. A ** node 0 ** runs the required services for the Agent-based Installer.
+
+.DHCP
+
+.. Preferred Method: `install-config.yaml` and `agent.config.yaml`
+
++
+You must specify only the value for the `rendezvousIP` field and `networkConfig` field must be left blank:
+
++
+.Sample agent-config.yaml.file
+
+[source,yaml]
+----
+  cat > agent-config.yaml << EOF
+  apiVersion: v1alpha1
+  kind: AgentConfig
+  metadata:
+    name: sno-cluster
+  rendezvousIP: 192.168.111.80 <1>
+----
+<1> The IP address for ** node 0**.
+
+.Static networking
+
+.. Preferred Method: `install-config.yaml` and `agent.config.yaml`
++
+.Scenario-1
+
+When you specify the value for the `rendezvousIP` and `networkConfig` fields, the `rendezvousIP` field is utilized.
+
++
+.Sample agent-config.yaml.file
++
+[source,yaml]
+----
+  cat > agent-config.yaml << EOF
+  apiVersion: v1alpha1
+  kind: AgentConfig
+  metadata:
+    name: sno-cluster
+  rendezvousIP: 192.168.111.80
+  hosts:
+    - hostname: master-0
+      interfaces:
+        - name: eno1
+          macAddress: 00:ef:44:21:e6:a5
+      networkConfig:
+        interfaces:
+          - name: eno1
+            type: ethernet
+            state: up
+            mac-address: 00:ef:44:21:e6:a5
+            ipv4:
+              enabled: true
+              address:
+                - ip: 192.168.111.80
+                  prefix-length: 23
+----
+
++
+.Scenario-2
+When you specify the values for the `networkConfig` field, the value from the `interfaces` field is utilized. This is when there is no `rendezvousIP` field.
++
+[source,yaml]
+----
+  cat > agent-config.yaml << EOF
+  apiVersion: v1alpha1
+  kind: AgentConfig
+  metadata:
+    name: sno-cluster
+  hosts:
+    - hostname: master-0
+      interfaces:
+        - name: eno1
+          macAddress: 00:ef:44:21:e6:a5
+      networkConfig:
+        interfaces:
+          - name: eno1
+            type: ethernet
+            state: up
+            mac-address: 00:ef:44:21:e6:a5
+            ipv4:
+              enabled: true
+              address:
+                - ip: 192.168.111.80
+                  prefix-length: 23
+----
++
+Note that the first IP is utilized for the IP address of ** node 0**.
```

Comment:
I think it is the lowest IP

Reply:
I think its a great idea to include this example. It should also include the stanzas for `dns-resolver` and `routes` in the example.

Reply:
@bfournie Added the resolver and routes section

Reply:
@celebdor I dont understand what you mean by the lowest IP , is it not the first IP value given?

Reply:
@celebdor  Is Agent-based Installer supporting all the five Modes? ref https://docs.openshift.com/container-platform/4.11/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-example-bond-nncp_k8s_nmstate-updating-node-network-config

Reply:
IPs are integer numbers, so the lowest means the IP that when represented as an integer, has the lowest value

Reply:
Regarding bonding modes, if they work with OpenShift as documented in 4.11, I don't see why they would not work for agent based installation

Reply:
ok noted

Reply:
Addressed lowest IP